packages/ros2/src/SystemHandle.cpp

```diff
@@ -130,25 +125,36 @@ bool SystemHandle::configure(
     std::string domain = domain_node.as<std::string>();
 
     success += set_platform_env("ROS_DOMAIN_ID", domain.c_str(), true);
+  }
 
-    _node = std::make_shared<rclcpp::Node>(name, ns);
-
-    if(previous_domain.empty())
-    {
-      success += unset_platform_env("ROS_DOMAIN_ID");
-    }
-    else
-    {
-      success += set_platform_env("ROS_DOMAIN_ID", previous_domain.c_str(), true);
-    }
+  // Initialize context after any change to the ROS_DOMAIN_ID env variable
+  rclcpp::InitOptions init_options;
+  // Quiet "logging initialized more than once" warnings
+  init_options.auto_initialize_logging(false);
+  // As of Galactic, we can also set the domain ID here instead of using the env variable
+  // init_options.set_domain_id(domain_node.as<size_t>());
```

Comment:
As a further refactor, we might be able to reach down to the `rcl` layer to configure the domain ID instead of using the environment variable.